### PR TITLE
ci: Update existing GitHub workflows

### PR DESCRIPTION
**Pull request by Margen67**
```
.ci_build_samples.sh:
 Use one less cd command.
 chmod +x
Upgrade checkout to v2.
Use -e when invoking shell(s).
Run .ci_build_samples.sh directly.
build_samples.yml:
 Don't use multi-line syntax for one command.
 Windows:
  Opt out of PowerShell telemetry.
  Use curl instead of Invoke-WebRequest.
 macOS:
  Move exported variables to env:.
  Opt out of brew analytics.
 Simplify ci_build_samples commands.
 Ubuntu:
  Remove unneeded packages.
  Remove unneeded PATH addition.
```

### CODE REVIEW

**Comment by JayFoxRox** [comment]
While you are at it, can you also do https://dev.to/github/running-github-actions-ci-cd-triggers-on-specific-branches-5e6m (or similar)?

I trigger 8 different builds when I push to my own repository while having a PR to my own repository: 4x for `push` and 4x for `pull_request`. That push event should probably only happen when pushing to master.

Feel free to ignore this request at your discretion, it can also be done at a different time.

File: .github/workflows/build_samples.yml [hunk 1]
```diff
@@ -9,70 +9,67 @@ jobs:
...
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
         with:
-          fetch-depth: 1
           submodules: recursive
       - name: Install Dependencies
         run: |
-          sudo apt-get update -y && sudo apt-get install -y flex bison clang llvm lld
+          sudo apt-get -y update
+          sudo apt-get -y install llvm lld
```
Comment (thrimbor) [inline .github/workflows/build_samples.yml:73]: What about the other dependencies?

File: .github/workflows/build_docker_image.yml [hunk 2]
```diff
@@ -12,9 +12,8 @@ jobs:
     timeout-minutes: 20
     steps:
       - name: Clone Tree
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
         with:
-          fetch-depth: 1
```
Comment (thrimbor) [inline .github/workflows/build_docker_image.yml:17]: Why the removal?

**Comment by thrimbor** [comment]
> I trigger 8 different builds when I push to my own repository while having a PR to my own repository: 4x for `push` and 4x for `pull_request`. That push event should probably only happen when pushing to master.

I would like to keep the push event, I rarely do self-PRs but still want to have my code built. Maybe introducing [a branch filter](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-ignoring-branches-and-tags) would be an option (something like skipping all branches with a `-noci` suffix)?

**Comment by JayFoxRox** [comment]
> I would like to keep the push event, I rarely do self-PRs but still want to have my code built.

Fair argument. I didn't think this through.

> Maybe introducing a branch filter would be an option

Not necessary; It just complicates stuff and I'd not use it.
I'll just ignore the redundant CI builds.

File: .github/workflows/build_docker_image.yml (hunk 2 above)
Comment (Margen67) [reply .github/workflows/build_docker_image.yml:17]: checkout v2 is fetch-depth 1 by default.

File: .github/workflows/build_samples.yml (hunk 1 above)
Comment (Margen67) [reply .github/workflows/build_samples.yml:73]: This is the minimum needed to build. flex and bison are already installed: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#installed-apt-packages

Comment (thrimbor) [reply .github/workflows/build_samples.yml:73]: My concern is that these are subject to change with newer runner releases, but I guess we cannot expect to be able to upgrade without breakage anyway.

**Comment by JayFoxRox** [comment]
I think this can be merged, but it needs the commit to be cleaned up:
- Prefix style
- Avoid long commit description (= we usually prefer a single line commit description)

**Comment by Margen67** [comment]
> 
> 
> I think this can be merged, but it needs the commit to be cleaned up:
> 
>     * Prefix style
> 
>     * Avoid long commit description (= we usually prefer a single line commit description)

Should I split each change so it only takes up 1-2 lines in the commit description?

**Comment by JayFoxRox** [comment]
Sorry for not getting back earlier. I try to minimize my involvement with Xbox stuff and was hoping someone else would respond.

> Should I split each change so it only takes up 1-2 lines in the commit description?

However you like. I don't mind a single commit for small changes like this, but it should follow the existing commit style (= prefix, no long description, commit title is punctual which is short enough to not cause a break on GitHub).

Renaming the existing commit to "ci: Update existing GitHub workflows" is totally fine *with me*.

**Comment by JayFoxRox** [comment]
Commit still has the long multi-line description. When I find time for it and it isn't removed I'll force push it away and merge this PR (unless there's a veto).

**Comment by thrimbor** [comment]
Pushed away the long description, merging now.

**Pull request merged**